ui/console/console.go

```diff
@@ -0,0 +1,238 @@
+package console
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/fatih/color"
+	"github.com/mattn/go-isatty"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/term"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Console enables synced writing to stdout and stderr ...
+type Console struct {
+	IsTTY          bool
+	outMx          *sync.Mutex
+	Stdout, Stderr OSFileW
+	Stdin          OSFileR
+	rawStdout      io.Writer
+	stdout, stderr *consoleWriter
+	theme          *theme
+	signalNotify   func(chan<- os.Signal, ...os.Signal)
+	signalStop     func(chan<- os.Signal)
+	logger         *logrus.Logger
+}
+
+// New returns the pointer to a new Console value.
+func New(
+	stdout, stderr OSFileW, stdin OSFileR,
+	colorize bool, termType string,
+	signalNotify func(chan<- os.Signal, ...os.Signal),
+	signalStop func(chan<- os.Signal),
+) *Console {
+	outMx := &sync.Mutex{}
+	outCW := newConsoleWriter(stdout, outMx, termType)
+	errCW := newConsoleWriter(stderr, outMx, termType)
+	isTTY := outCW.isTTY && errCW.isTTY
```

Comment:
Note that this mostly assumes that both stdout and stderr will either be a TTY or not together. I.e. that there won't be a case where stdout is a TTY, but stderr isn't. Previously we were checking this separately.

Is this a safe assumption to make? In that case, I might just pull out the check from `newConsoleWriter()` and do it only once here, for either stdout or stderr, and remove the `isTTY` field from `consoleWriter`.

WDYT?